app/controllers/api/vms_controller.rb

```diff
@@ -351,7 +351,7 @@ def vm_event(vm, event_type, event_message, event_time)
     def retire_vm(vm, id, data)
       desc = "#{vm_ident(vm)} retiring"
       desc << " on #{data['date']}" if Hash(data)['date'].present?
-      retire_resource(:vms, id, data)
+      method(:retire_resource).super_method.call(:vms, id, data)
```

Comment:
`retire_resource` here calls the method that invokes this code, hence infinite recursion. So we need to call the super method. Ugly though, I agree.

< If not, perhaps it would feel less clunky to alias the parent's method (perhaps to retire_resource_vms or direct_retire_resource) and then call that alias directly?

Possibly, but then I'm moving away from the polymorphism I"m aiming for in this PR. I considered moving things around (I guess inlining this method) so that I could just call `super` in `VmsController#retire_resource`, but didn't because I was already making a big change. I'm happy to tack on a commit to tidy that up a bit though. Thoughts with going with this approach?